From 3e380acf5366cf7e907aeedc9af9fa831178e31b Mon Sep 17 00:00:00 2001
From: Matthieu Herrb <matthieu.herrb@laas.fr>
Date: Sun, 14 Nov 2021 17:38:24 +0100
Subject: [PATCH] Don't set SourceValidate pointer to NULL

inspired from similar code in amdgpu, fixes a crash when xrandr(1)
is invoqued with X server 21.1.1

Signed-off-by: Matthieu Herrb <matthieu.herrb@laas.fr>
---
 src/radeon_kms.c | 9 ++++++++-
 1 file changed, 8 insertions(+), 1 deletion(-)

diff --git a/src/radeon_kms.c b/src/radeon_kms.c
index 62962d61..51b00dd9 100644
--- a/src/radeon_kms.c
+++ b/src/radeon_kms.c
@@ -931,6 +931,13 @@ radeon_dirty_update(ScrnInfoPtr scrn)
 	}
 }
 
+static void
+radeonSourceValidate(DrawablePtr draw, int x, int y, int w, int h,
+		     unsigned int subWindowMode)
+{
+}
+
+
 
 Bool
 radeon_scanout_do_update(xf86CrtcPtr xf86_crtc, int scanout_id,
@@ -993,7 +1000,7 @@ radeon_scanout_do_update(xf86CrtcPtr xf86_crtc, int scanout_id,
 	    SetPicturePictFilter(src, xf86_crtc->filter, xf86_crtc->params,
 				 xf86_crtc->nparams);
 
-	pScreen->SourceValidate = NULL;
+	pScreen->SourceValidate = radeonSourceValidate;
 	CompositePicture(PictOpSrc,
 			 src, NULL, dst,
 			 extents.x1, extents.y1, 0, 0, extents.x1,
-- 
GitLab

